finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4DispatcherBase.scala

```diff
@@ -87,13 +87,13 @@ trait Netty4DispatcherBase[SendMsg <: Message, RecvMsg <: Message] {
             case rst: Reset => rst
             case _ => Reset.Cancel
           }
-          if (!closed.get) { writer.reset(id, rst); () }
+          if (!closed.get) { writer.reset(H2FrameStream(id, Http2Stream.State.CLOSED), rst); () }
```

Comment:
Are we confident that CLOSED is the right state here rather than one of the half-closed states?  Asking because I honestly have no idea.  (likewise for the other reset calls in this file)

Reply:
According to the state transition diagram in http://httpwg.org/specs/rfc7540.html#StreamStates, I _believe_ sending or receiving a reset should always transition the stream to CLOSED.

TIOLI: if CLOSED is the only valid transition on a reset, consider encapsulating `writer.reset` in another function that takes an `id` and always constructs a CLOSED stream to pass to `writer.reset`....

Reply:
That's a great idea! I could just change `writer.reset` to accept only and id and just generate a stream object that is in the `CLOSED` state. My only concern is that the method signature in `NettyH2Writer` would look different than i.e. `writer.reset` only accepts an id while `write(...)` accepts a `H2FrameStream` and other parameters.